### PR TITLE
TKX-1096 Custom timeout, simplified configuration, drop T4 transform, dependabot groups

### PR DESCRIPTION
**Pull request by emanuelgaspar**
Allow `HttpClient.Timeout` to be customised for each client.
The .NET default is 100 seconds, which seems excessive:
<https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.timeout?view=net-7.0#:~:text=The%20default%20value%20is%20100%2C000%20milliseconds%20(100%20seconds).>

Simplify settings, use `TimeSpan` and defaults.
Remove T4 transformation.
`dependabot` groups

Nuget updates, `common` reuse where possible.
All tests working.

## IMPORTANT
Before making changes to `CoinGeckoApiConfiguration`, I checked all projects using it.
No external project was setting custom values for the properties converted to `TimeSpan` in this PR.
The conversion to `Uri` is also fine, I added a small unit test to sanity check, just to be sure.

### CODE REVIEW

**Review by joaofbvieira — DISMISSED** [review]
LGTM

**Review by monsieurleberre — APPROVED** [review]
looks better thanks

File: src/Trakx.CoinGecko.ApiClient/Configuration/ApiClientExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+public static partial class ApiClientExtensions
+{
+    public static IServiceCollection AddCoinGeckoClient(
+        this IServiceCollection services, IConfiguration configuration)
+    {
+        var section = configuration.GetSection(nameof(CoinGeckoApiConfiguration));
+        services.Configure<CoinGeckoApiConfiguration>(section);
+
+        var typedConfig = section.Get<CoinGeckoApiConfiguration>()!;
```
Comment (monsieurleberre) [inline src/Trakx.CoinGecko.ApiClient/Configuration/ApiClientExtensions.cs:26]: Could you have used your 
Trakx.Common.Configuration.ConfigurationExtensions.GetConfiguration<<TConfiguration>>(this IConfiguration configuration) ?

Comment (emanuelgaspar) [reply src/Trakx.CoinGecko.ApiClient/Configuration/ApiClientExtensions.cs:26]: Ah, yes, I'll add it in the next PR.

**Pull request merged**